pkg/operation/botanist/constraints_check.go

```diff
@@ -0,0 +1,133 @@
+	validatingWebhookConfigs := &admissionregistrationv1beta1.ValidatingWebhookConfigurationList{}
+	if err := b.K8sShootClient.Client().List(ctx, validatingWebhookConfigs); err != nil {
+		return nil, fmt.Errorf("could not get ValidatingWebhookConfigurations of Shoot cluster to check if Shoot can be hibernated")
+	}
+
+	for _, webhookConfig := range validatingWebhookConfigs.Items {
+		if exitConstraint := checker.CheckWebhooks(constraint, "ValidatingWebhookConfiguration", webhookConfig.Name, webhookConfig.Webhooks); exitConstraint != nil {
+			return exitConstraint, nil
+		}
+	}
+
+	mutatingWebhookConfigs := &admissionregistrationv1beta1.MutatingWebhookConfigurationList{}
+	if err := b.K8sShootClient.Client().List(ctx, mutatingWebhookConfigs); err != nil {
+		return nil, fmt.Errorf("could not get MutatingWebhookConfigurations of Shoot cluster to check if Shoot can be hibernated")
+	}
+
+	for _, webhookConfig := range mutatingWebhookConfigs.Items {
+		if exitConstraint := checker.CheckWebhooks(constraint, "MutatingWebhookConfiguration", webhookConfig.Name, webhookConfig.Webhooks); exitConstraint != nil {
+			return exitConstraint, nil
+		}
+	}
+
+	c := gardencorev1alpha1helper.UpdatedCondition(constraint, gardencorev1alpha1.ConditionTrue, "NoProblematicWebhooks", "Shoot can be hibernated.")
+	return &c, nil
+}
+
+// CheckWebhooks checks all webhooks in the Shoot to determine if hibernation is not possible due to problematic Webhooks.
+// Problematic webhooks are: ValidationWebhookConfigurations and MutatingWebhookConfigurations with rules
+// for CREATE pods or nodes and failurePolicy=Fail and webhooks[].clientConfig.url == nil.
+// Otherwise, we can never wake up this shoot cluster again as new nodes cannot get created,
+// or our system component pods cannot get created (because the validating webhook pod is not yet running).
+func (c *ConstraintsChecker) CheckWebhooks(constraint gardencorev1alpha1.Condition, configKind string, configName string, webhooks []admissionregistrationv1beta1.Webhook) *gardencorev1alpha1.Condition {
+	for _, webhook := range webhooks {
+		if webhook.FailurePolicy != nil && *webhook.FailurePolicy != admissionregistrationv1beta1.Fail {
+			// in admissionregistration.k8s.io/v1 FailurePolicy is defaulted to `Fail`
+			// see https://github.com/kubernetes/api/blob/release-1.16/admissionregistration/v1/types.go#L195
+			// and https://github.com/kubernetes/api/blob/release-1.16/admissionregistration/v1/types.go#L324
+			// therefore, don't ignore FailurePolicy==nil
+			continue
+		}
+
+		for _, rule := range webhook.Rules {
+			apiGroups := sets.NewString(rule.APIGroups...)
+			resources := sets.NewString(rule.Resources...)
+
+			if apiGroups.Has(corev1.GroupName) && resources.HasAny("pods", "nodes") {
+				for _, op := range rule.Operations {
+					if op == admissionregistrationv1beta1.Create || op == admissionregistrationv1beta1.OperationAll || (resources.Has("nodes") && op == admissionregistrationv1beta1.Update) {
```

Comment:
Why only UPDATE for nodes? Did you test this scenario and it worked? I'm not sure if it would work, hence, I would simply include UPDATE for both pods and nodes.

Reply:
Oh, sorry. There was a misunderstanding from my side. You are right, I didn't consider this case...
I changed the check to consider Operations `CREATE`, `UPDATE` and `*` on both Pods and Nodes to be problematic.
Also rebased and solved the conflicts, PTAL.